src/Button/styles.ts

```diff
@@ -170,96 +172,140 @@ export const getVariantStyles = (variant: VariantType = 'default', theme?: Theme
+  '&[href]': {
+    display: 'inline-flex',
+    '&:hover': {
+      textDecoration: 'none',
+    },
+  },
+  '&:hover': {
+    transitionDuration: '80ms',
   },
-  '&:disabled svg': {
-    opacity: '0.6',
+  '&:active': {
+    transition: 'none',
+  },
+  '&:disabled': {
+    cursor: 'not-allowed',
+    boxShadow: 'none',
   },
   '@media (forced-colors: active)': {
     '&:focus': {
       // Support for Windows high contrast https://sarahmhigley.com/writing/whcm-quick-tips
       outline: 'solid 1px transparent',
     },
   },
+  '[data-component=ButtonCounter]': {
+    fontSize: '1',
+  },
+  '&[data-component=IconButton]': {
+    display: 'inline-grid',
+    padding: 'unset',
+    placeContent: 'center',
+    width: '32px',
+  },
+  '&[data-size="small"]': {
+    padding: '0 8px',
+    height: '28px',
+    gap: '4px',
+    fontSize: '0',
+
+    '[data-component="text"]': {
+      lineHeight: 'calc(20 / 12)',
+    },
+
+    '[data-component=ButtonCounter]': {
+      fontSize: '0',
+    },
+
+    '[data-component="buttonContent"] > :not(:last-child)': {
+      mr: '4px',
+    },
+
+    '&[data-component=IconButton]': {
+      width: '28px',
+      padding: 'unset',
+    },
+  },
+  '&[data-size="large"]': {
+    padding: '0 16px',
+    height: '40px',
+    gap: '8px',
+
+    '[data-component="buttonContent"] > :not(:last-child)': {
+      mr: '8px',
+    },
+
+    '&[data-component=IconButton]': {
+      width: '40px',
+      padding: 'unset',
+    },
+  },
 })
 
 export const getButtonStyles = (theme?: Theme) => {
   const styles = {
     ...getBaseStyles(theme),
-    display: 'grid',
-    gridTemplateAreas: '"leadingIcon text trailingIcon"',
-    '& > :not(:last-child)': {
-      mr: '2',
+    '&[data-component="block"]': {
+      width: '100%',
     },
-    '[data-component="leadingIcon"]': {
-      gridArea: 'leadingIcon',
+    '[data-component="leadingVisual"]': {
+      gridArea: 'leadingVisual',
     },
     '[data-component="text"]': {
       gridArea: 'text',
+      lineHeight: 'calc(20/14)',
+      whiteSpace: 'nowrap',
+    },
+    '[data-component="trailingVisual"]': {
+      gridArea: 'trailingVisual',
+    },
+    '[data-component="trailingAction"]': {
+      marginRight: '-4px',
     },
-    '[data-component="trailingIcon"]': {
-      gridArea: 'trailingIcon',
+    '[data-component="buttonContent"]': {
+      flex: '1 0 auto',
+      display: 'grid',
+      gridTemplateAreas: '"leadingVisual text trailingVisual"',
+      gridTemplateColumns: 'min-content minmax(0, auto) min-content',
+      alignItems: 'center',
+      alignContent: 'center',
+    },
+    '[data-component="buttonContent"] > :not(:last-child)': {
+      mr: '8px',
     },
   }
   return styles
 }
+
+export const getAlignContentSize = (alignContent: AlignContent = 'center') => {
+  const style = {
+    center: {
+      justifyContent: 'center',
+    },
+    start: {
+      justifyContent: 'flex-start',
+    },
+  }
+  return style[alignContent]
+}
```

Comment:
Since we only have two possible values that `alignContent` could be, we could simplify this.

```suggestion
export const getAlignContentSize = (alignContent: AlignContent = 'center') => ({
  justifyContent: alignContent === 'center' ? 'center' : 'flex-start' 
})
```